cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -1557,6 +1558,35 @@ impl ManagedEnvironment {
 
         Ok(PullResult::Updated)
     }
+
+    /// Detach the environment from the remote repository.
+    ///
+    /// And return a [PathEnvironment] representing
+    /// the current local state of the environment.
+    ///
+    /// This method converts the [ManagedEnvironment] into a [PathEnvironment]
+    /// and deletes the corresponding branch in the [FloxMeta] repository.
+    ///
+    /// **The remote repository is not affected.**
+    pub fn into_path_environment(self, flox: &Flox) -> Result<PathEnvironment, EnvironmentError> {
+        let _ = self.local_env_or_copy_current_generation(flox)?;
+
+        let branch = branch_name(&self.pointer, &self.path);
+        self.floxmeta.git.delete_branch(&branch, true).unwrap();
+
+        fs::remove_file(self.path.join(GENERATION_LOCK_FILENAME))
+            .map_err(ManagedEnvironmentError::WriteLock)?;
+
+        let path_pointer = PathPointer::new(self.name());
+        fs::write(
+            self.path.join(ENVIRONMENT_POINTER_FILENAME),
+            serde_json::to_string(&path_pointer)
+                .map_err(ManagedEnvironmentError::SerializePointer)?,
+        )
+        .map_err(ManagedEnvironmentError::WritePointer)?;
+
+        PathEnvironment::new(self.path, path_pointer)
```

Comment:
suggestion blocking: re-use/factor out ManagedEnvironment::delete and PathEnvironment::open so we
- deregister the managed env
- register the path env
- remove the managed env gcroot

Should we build and create a gcroot for the path env as well?